services/gossip/codec/topic_header_sync_test.go

```diff
@@ -0,0 +1,305 @@
+	test.RequireCmpEqual(t, message, decoded, "decoded encoded should equal to original")
+}
+
+func TestHeaderSync_EmptyHeaderSyncRequest(t *testing.T) {
+	_, err := DecodeHeaderSyncRequest(builders.EmptyPayloads(2))
+	require.Error(t, err, "decode should fail and return error")
+}
+
+func TestHeaderSync_HeaderSyncRequestDoNotFailWhenSenderContainsNil(t *testing.T) {
+	message := &gossipmessages.HeaderSyncRequestMessage{
+		SignedChunkRange: (&gossipmessages.HeaderSyncRangeBuilder{
+			HeaderType:                gossipmessages.HEADER_TYPE_RESULTS_BLOCK_HEADER_WITH_PROOF,
+			FirstBlockHeight:         1001,
+			LastBlockHeight:          2001,
+			LastCommittedBlockHeight: 3001,
+		}).Build(),
+		Sender: (&gossipmessages.SenderSignatureBuilder{
+			SenderNodeAddress: nil,
+			Signature:         nil,
+		}).Build(),
+	}
+
+	payloads, err := EncodeHeaderSyncRequest((&gossipmessages.HeaderBuilder{}).Build(), message)
+	require.NoError(t, err, "encode should not fail")
+	decoded, err := DecodeHeaderSyncRequest(payloads[1:])
+	require.NoError(t, err, "decode should not fail")
+	test.RequireCmpEqual(t, message, decoded, "decoded encoded should equal to original")
+	test.RequireDoesNotContainNil(t, decoded)
+}
+
+func TestHeaderSync_HeaderSyncResponse(t *testing.T) {
+	message := &gossipmessages.HeaderSyncResponseMessage{
+		SignedChunkRange: (&gossipmessages.HeaderSyncRangeBuilder{
+			HeaderType:               gossipmessages.HEADER_TYPE_RESULTS_BLOCK_HEADER_WITH_PROOF,
+			FirstBlockHeight:         1001,
+			LastBlockHeight:          2001,
+			LastCommittedBlockHeight: 3001,
+		}).Build(),
+		Sender: (&gossipmessages.SenderSignatureBuilder{
+			SenderNodeAddress: []byte{0x01, 0x02, 0x03},
+			Signature:         []byte{0x04, 0x05, 0x06},
+		}).Build(),
+		HeaderWithProof: []*gossipmessages.ResultsBlockHeaderWithProof{
+			buildResultsBlockHeaderWithProof(3),
+			buildResultsBlockHeaderWithProof(4),
+		},
+	}
+
+	payloads, err := EncodeHeaderSyncResponse((&gossipmessages.HeaderBuilder{}).Build(), message)
+	require.NoError(t, err, "encode should not fail")
+	decoded, err := DecodeHeaderSyncResponse(payloads[1:])
+	require.NoError(t, err, "decode should not fail")
+	test.RequireCmpEqual(t, message, decoded, "decoded encoded should equal to original")
+}
+
+
+type corruptBlockPair struct {
+	txContainer *protocol.TransactionsBlockContainer
+	rxContainer *protocol.ResultsBlockContainer
+}
+
+func CorruptBlockPair() *corruptBlockPair {
+	return &corruptBlockPair{}
+}
+
+func (c *corruptBlockPair) Build() *protocol.BlockPairContainer {
+	return &protocol.BlockPairContainer{
+		TransactionsBlock: c.txContainer,
+		ResultsBlock:      c.rxContainer,
+	}
+}
+
+
+
+func buildResultsBlockHeaderWithProof(height primitives.BlockHeight) *gossipmessages.ResultsBlockHeaderWithProof {
+	empty32ByteHash := hash.Make32EmptyBytes()
+	rxBlockHeader := (&protocol.ResultsBlockHeaderBuilder{
+		ProtocolVersion:                 primitives.ProtocolVersion(1),
```

Comment:
please don't use this PV ... once we have merge with refTime use config.MAXIMAL_PROTOCOL_VERSION_SUPPORTED_VALUE